### PR TITLE
Enhancements to Pagination and Other Fixes

### PR DESCRIPTION
**Pull request by Ashesh3**
This pull request includes enhancements to pagination in the codebase. It improves the pagination functionality in various components and fixes some UI issues related to pagination.

Also adds pagination to test suite questions list page

### CODE REVIEW

File: src/app/(main)/client.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { useEffect } from "react";
 
 export default function Client() {
     const projectsQuery = useQuery<paginatedResponse<Project>>({
-        queryKey: ["projects"],
+        queryKey: ["app:projects"],
```
Comment (skks1212) [inline src/app/(main)/client.tsx:10]:
```suggestion
        queryKey: ["projects", "app"],
```
Recommended convention by react query

File: src/app/admin/page.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { useEffect, useState } from "react";
 export default function Page() {
   const [isArchived, setIsArchived] = useState(false);
   const projectsQuery = useInfiQuery({
-    queryKey: ["projects"],
+    queryKey: ["admin:projects"],
```
Comment (skks1212) [inline src/app/admin/page.tsx:12]:
```suggestion
    queryKey: ["projects", "admin"],
```

**Review by skks1212 — APPROVED** [review]

**Comment by skks1212** [comment]
@Ashesh3 can you make the changes? 😀

**Comment by Ashesh3** [comment]
> @Ashesh3 can you make the changes? 😀

Done!

**Pull request merged**